br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -87,8 +195,15 @@ func NewDuplicateManager(
 	}, nil
 }
 
-func (manager *DuplicateManager) CollectDuplicateRowsFromTiKV(ctx context.Context, tbl table.Table) error {
-	log.L().Info("Begin collect duplicate data from remote TiKV")
+// CollectDuplicateRowsFromTiKV collects duplicated rows already imported into TiKV.
+//
+// Collection result are saved into the ErrorManager.
+func (manager *DuplicateManager) CollectDuplicateRowsFromTiKV(ctx context.Context, tbl table.Table) (err error) {
+	logTask := log.L().Begin(zapcore.InfoLevel, "collect duplicate data from remote TiKV")
+	defer func() {
+		logTask.End(zapcore.InfoLevel, err)
+	}()
+
```

Comment:
BTW, shall we need to restrict the concurrency in L221?

Reply:
> Do we need to use the same session vars in `kv.NewTableKVDecoder` at L212? 🤔

what do you mean?

Reply:
https://github.com/pingcap/tidb/blob/bb79bb7659e02a678fee534f2dc1b8b82701d357/br/pkg/lightning/backend/local/duplicate.go#L97-L99

The code location has changed due to new commit😂. Do we need to set the correct session vars to get the correct result in `TableKVDecoder`? Though this logic is not introduced in this PR😅.

Reply:
@glorv AFAIK the session options do not affect how the raw KV are decoded.

Reply:
https://github.com/pingcap/tidb/blob/48c624b201ee6629367232b2229f6e0cc0defa07/table/tables/tables.go#L988-L991

The `DecodeRawRowData` method called by `TableKVDecoder` depends on the timezone setting. Because we have set the timezone in `TableKVEncoder` from the tidb cluster, we should also use the same value in decoder.

Reply:
OK.

since this only affects report of TIMESTAMP type I'm going to just add a TODO 😂